dds_cli/project_status.py

```diff
@@ -148,6 +148,112 @@ def update_status(self, new_status, deadline=None, is_aborted=False, no_mail=Fal
+            params={"project": self.project},
+            json=extra_params,
+        )
+
+        # Structure of the response:
+        #   {
+        #   'default_unit_days': 30,
+        #   'project_info': {
+        #       'Created by': 'First Unit User',
+        #       'Description': 'This is a test project',
+        #       'Last updated': 'Wed, 18 Oct 2023 08:40:43 GMT',
+        #       'PI': 'support@example.com',
+        #       'Project ID': 'project_1',
+        #       'Size': 0,
+        #       'Status': 'Available',
+        #       'Title': 'First Project'
+        #       },
+        #   'project_status': {
+        #       'current_deadline': 'Sat, 04 Nov 2023 23:59:59 GMT',
+        #       'current_status': 'Available'},
+        #   'warning': 'Operation must be confirmed before proceding.'
+        #   }
+
+        # Extract default unit days and current deadline
+        default_unit_days = response_json.get("default_unit_days")
+        current_deadline = response_json.get("project_status").get("current_deadline")
+
+        # print information about the project status and table with the project info
+        print_info = (
+            f"\nCurrent deadline: [b][green]{current_deadline}[/green][/b]\n"
+            f"Default deadline extension: [b][green]{default_unit_days}[/green][/b] days\n"
+        )
+        table = self.generate_project_table(project_info=response_json.get("project_info"))
+        dds_cli.utils.console.print(table)
+        dds_cli.utils.console.print(print_info)
+
+        # First question, number of days to extend the deadline
+        prompt_question = (
+            f"Enter the number of days you want to extend the project, "
+            f"the number of days has to be equal or same as "
+            f"[b][green]{default_unit_days}[/green][/b].\n"
+            f"Or leave it empty to apply the default "
+            f"[b][green]{default_unit_days} days [/green][/b]"
+        )
+
+        dds_cli.utils.console.print(prompt_question)
+        extend_deadline = rich.prompt.Prompt.ask("-")
+        if not extend_deadline:
+            # Set extend_deadline to default
+            extend_deadline = default_unit_days
+        try:
+            # the input was an string --> convert to integer
+            extend_deadline = int(extend_deadline)
+            if extend_deadline > default_unit_days:
+                dds_cli.utils.console.print(
+                    "\n[b][red]The number of days has to be lower than "
+                    "the default deadline extension number[/b][/red]\n"
+                )
+                LOG.info("Exiting the function, try again")
+                sys.exit(0)
+
+        except ValueError:
+            dds_cli.utils.console.print(
+                "\n[b][red]Remember to write the number of days "
+                "using numbers (dont use letters)[/b][/red]\n"
+            )
+            LOG.info("Exiting the function, try again")
+            sys.exit(0)
```

Comment:
Raise exceptions instead so that it's caught in `__main__.py`
```suggestion
        try:
            # the input was an string --> convert to integer
            extend_deadline = int(extend_deadline)
            if extend_deadline > default_unit_days:
                raise DDSCLIException(                    
                    "\n[b][red]The number of days has to be lower than or equal to your unit's default: {default_unit_days}[/b][/red]\n"
                )
        except ValueError:
            raise DDSCLIException(
                "\n[b][red]Invalid value. Remember to enter a digit (not letters) when being asked for the number of days.[/b][/red]\n"
            )
```

Reply:
On second thought, I am thinking that it actually might be better to add those two checks to the api. Maybe the client is not the place to handled them. What do you think?

Reply:
Yeah I agree, even better 👍🏻